woocommerce/woocommerce-android
# Issue/4660 add support for optional firmware updates

Closes: #4660 


### Description
This PR adds support for optional firmware updates for Stripe's new SDK version V2. You can read more about optional updates [here](https://stripe.com/docs/terminal/readers/bbpos-chipper2xbt#optional-updates)


### Testing instructions
1. Go to `Settings` from the `MyStore` tab
2. Click on `In-Person Payments` under the `Store settings` option
3. Click on the `Manage Card Reader` option in the `In-Person Payments` screen
4. Click on the `Connect card reader` button
5. After successful connection, see that there will be a software update available banner and a button to update.
6. Click on it, you should see a progress dialog updating the software
7. Once the update is complete successfully, you no longer see the update banner in the `card reader detail screen`.
8. Come back from the `card reader detail screen` and enter the screen again just to make sure the card reader is in the proper state
9. Also, make sure to check the card reader will be in the proper state when there is `no update available`. You can check this by changing Simulator configuration to `SimulateReaderUpdate.NONE` in `TerminalWrapper.kt` file. Line no 76


### Images/gif

https://user-images.githubusercontent.com/1331230/131426514-b0a040de-d5f8-4b15-848c-ea7d5c517670.mp4
